[CI] GenerativeMetricsIT test failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #10697 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/uuib6m33363oe)
- [elasticsearch-periodic-platform-support #10697 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/pcdiupxaeedca)
- [elasticsearch-periodic-platform-support #10697 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/lcchr22uaqx44)
- [elasticsearch-periodic #10707 / openjdk21_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/ppkiqjlai2kdy)
- [elasticsearch-pull-request #92764 / part-3](https://gradle-enterprise.elastic.co/s/tnqdzeadyspnu)
- [elasticsearch-pull-request #92480 / part-3](https://gradle-enterprise.elastic.co/s/zw5xqtsltssj6)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeMetricsIT.test" -Dtests.seed=AE93AEA84A14A6A4 -Dtests.locale=mn-MN -Dtests.timezone=Africa/Johannesburg -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.GenerativeMetricsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: ts k8s*,k8s-downsamp*,k8s-downsampled | grok event "%{WORD:event} %{WORD:UNTlbZFA}" | dissect UNTlbZFA "%{network.total_bytes_in} %{PmrvMrPKrMaw}" | keep `network.bytes_in`, cluster, `events_received`, `network.total_bytes_out`, *, client.ip, event_city_boundary, `@timestamp`, PmrvMrPKrMaw, `network.eth0.rx`, `network.cost`, event_shape, `event`, `event_city_boundary`, network.eth0.firmware_version, `network.eth0.up` | stats  `network.total_cost` = min(network.bytes_in) by events_received, ISMGHYMewu = bucket(@timestamp,1hour) | limit 1682 | limit 9496 | keep ISMGHYMewu | limit 12925 | eval  CNofcFOnm = false, GiJPPsbrfA = null, ISMGHYMewu = -2856975220971678219, ImoefjbpY = false, uGwbVBGxZESC = -3200463956386419530, TnxgHfCdfp = -5875752450925686837, LbhcJnBPvVZ = "ST", iAFkPUpdFwqF = -6072868724081692699
exception: Connection is closed
```

**Issue Reasons:**
- [main] 3 consecutive failures in test test
- [main] 6 failures in test test (9.0% fail rate in 67 executions)
- [main] 2 failures in step part-3 (5.0% fail rate in 40 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (5.0% fail rate in 40 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -429,9 +429,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
   method: test {csv-spec:stats.CountDistinctWithConditions}
   issue: https://github.com/elastic/elasticsearch/issues/134993
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeMetricsIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/135055
 - class: org.elasticsearch.xpack.esql.heap_attack.HeapAttackIT
   method: testLookupExplosionBigString
   issue: https://github.com/elastic/elasticsearch/issues/135122

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
@@ -81,7 +81,9 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
         "time-series.*the first aggregation.*is not allowed",
         "count_star .* can't be used with TS command",
         "time_series aggregate.* can only be used with the TS command",
-        "implicit time-series aggregation function .* doesn't support type .*"
+        "implicit time-series aggregation function .* doesn't support type .*",
+        "INLINE STATS .* can only be used after STATS when used with TS command",
+        "cannot group by a metric field .* in a time-series aggregation"
     );
 
     public static final Set<Pattern> ALLOWED_ERROR_PATTERNS = ALLOWED_ERRORS.stream()

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/generator/EsqlQueryGenerator.java
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/generator/EsqlQueryGenerator.java
@@ -133,9 +133,7 @@ public class EsqlQueryGenerator {
             if (executor.currentSchema().isEmpty()) {
                 break;
             }
-            commandGenerator = isTimeSeries && canGenerateTimeSeries
-                ? randomMetricsPipeCommandGenerator()
-                : randomPipeCommandGenerator();
+            commandGenerator = isTimeSeries && canGenerateTimeSeries ? randomMetricsPipeCommandGenerator() : randomPipeCommandGenerator();
             if (isTimeSeries) {
                 if (commandGenerator.equals(ForkGenerator.INSTANCE)) {
                     // don't fork with TS command until this is resolved: https://github.com/elastic/elasticsearch/issues/136927

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/generator/EsqlQueryGenerator.java
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/generator/EsqlQueryGenerator.java
@@ -133,28 +133,26 @@ public class EsqlQueryGenerator {
             if (executor.currentSchema().isEmpty()) {
                 break;
             }
-            boolean commandAllowed = false;
-            while (commandAllowed == false) {
-                commandGenerator = isTimeSeries && canGenerateTimeSeries
-                    ? randomMetricsPipeCommandGenerator()
-                    : randomPipeCommandGenerator();
-                if (isTimeSeries == false) {
-                    commandAllowed = true;
-                } else {
-                    if (commandGenerator.equals(TimeSeriesStatsGenerator.INSTANCE) || commandGenerator.equals(StatsGenerator.INSTANCE)) {
-                        if (canGenerateTimeSeries) {
-                            canGenerateTimeSeries = false;
-                            commandAllowed = true;
-                        }
-                    } else if (commandGenerator.equals(RenameGenerator.INSTANCE)) {
-                        // https://github.com/elastic/elasticsearch/issues/134994
-                        canGenerateTimeSeries = false;
-                        commandAllowed = true;
-                    } else {
-                        commandAllowed = true;
+            commandGenerator = isTimeSeries && canGenerateTimeSeries
+                ? randomMetricsPipeCommandGenerator()
+                : randomPipeCommandGenerator();
+            if (isTimeSeries) {
+                if (commandGenerator.equals(ForkGenerator.INSTANCE)) {
+                    // don't fork with TS command until this is resolved: https://github.com/elastic/elasticsearch/issues/136927
+                    continue;
+                }
+                if (commandGenerator.equals(TimeSeriesStatsGenerator.INSTANCE) || commandGenerator.equals(StatsGenerator.INSTANCE)) {
+                    if (canGenerateTimeSeries == false) {
+                        // Don't generate multiple stats commands in a single query for TS
+                        continue;
                     }
+                    canGenerateTimeSeries = false;
+                } else if (commandGenerator.equals(RenameGenerator.INSTANCE)) {
+                    // don't allow stats after a rename until this is resolved: https://github.com/elastic/elasticsearch/issues/134994
+                    canGenerateTimeSeries = false;
                 }
             }
+
             desc = commandGenerator.generate(executor.previousCommands(), executor.currentSchema(), schema, queryExecutor);
             if (desc == CommandGenerator.EMPTY_DESCRIPTION) {
                 continue;

